server/s3_object_list_response.cc

```diff
@@ -295,9 +296,15 @@ std::string& S3ObjectListResponse::get_multiupload_xml() {
   }
 
   for (auto&& object : object_list) {
+    std::string object_name = object->get_object_name();
+    if (chop_uploadid) {
+      // In case of multipart listing, chop uploadid from key
+      std::size_t pos = object_name.find("|");
+      object_name = object_name.substr(0, pos);
```

Comment:
can 'pos' variable have a not valid value, like std::npos ?

Reply:
Also we can have "|" in object name, right ?

Do we need handling of this scenario as well ?

Reply:
# aws s3api list-objects --bucket mybk
{
            "LastModified": "2021-04-01T11:50:56.000Z", 
            "ETag": "\"8c9e4b10ec6e7576c2c427f7fb89b58a\"", 
            "StorageClass": "STANDARD", 
            **"Key": "my_o|bj",** 
            "Owner": {
                "DisplayName": "new-acc", 
                "ID": "da9a0c3acd48427fbf95f8d3dfe5be182f52ca5d7b44413cb74ec394036f8e69"
            }, 
            "Size": 108
        }

Reply:
if pos is std::npos then substr will show complete string, I have put in check to call substr based on pos value

Since object name can have | (as pr aws s3 its suppose to be avoided, but not restricted), hence changed find() to rfind()